numpy/core/src/multiarray/ctors.c

```diff
@@ -1916,6 +1921,14 @@ PyArray_FromArray(PyArrayObject *arr, PyArray_Descr *newtype, int flags)
            !PyArray_EquivTypes(oldtype, newtype);
 
     if (copy) {
+
+        if( flags & NPY_ARRAY_ENSURENOCOPY ) {
+            PyErr_SetString(PyExc_RuntimeError,
```

Comment:
What to raise, `ValueError` or `RuntimeError`?

Reply:
I think a `ValueError` is more appropriate since the problem is the input value.

Reply:
I gave a thought to it and I agree with you and @seberg that the problem is with the combination of input values entered by the user. I will change this. In fact, I think `ValueError` seems more relatable in case deep copy is performed for scalars when creating an array.